components/server/src/workspace/config-provider.ts

```diff
@@ -43,53 +43,64 @@ export class ConfigProvider {
-            const definitelyGpConfig = this.fetchExternalGitpodFileContent({ span }, commit.repository);
-            const inferredConfig = this.inferingConfigProvider.infer(user, commit);
-
-            let customConfigString = await contextRepoConfig;
-            let configBasePath = '';
-            let fromDefinitelyGp = false;
-            if (!customConfigString) {
-                /* We haven't found a Gitpod configuration file in the context repo - check definitely-gp.
-                *
-                * In case we had found a config file here, we'd still be checking the definitely GP repo, just to save some time.
-                * While all those checks will be in vain, they should not leak memory either as they'll simply
-                * be resolved and garbage collected.
-                */
-                const { content, basePath } = await definitelyGpConfig;
-                customConfigString = content;
-                // We do not only care about the config itself but also where we got it from
-                configBasePath = basePath;
-                fromDefinitelyGp = true;
-            }
-
             let customConfig: WorkspaceConfig | undefined;
-            const logContext: LogContext = { userId: user.id };
-            if (customConfigString) {
+            if (AdditionalContentContext.is(commit) && commit.additionalFiles['.gitpod.yml']) {
+                const customConfigString = commit.additionalFiles['.gitpod.yml'];
                 const parseResult = this.gitpodParser.parse(customConfigString);
-                customConfig = parseResult.config
-                await this.fillInDefaultLocations(customConfig, inferredConfig);
+                customConfig = parseResult.config;
+                customConfig._origin = 'context';
                 if (parseResult.validationErrors) {
-                    log.error(logContext, `Skipping invalid config. Errors are ${parseResult.validationErrors.join(',')}`, { repoCloneUrl: commit.repository.cloneUrl, revision: commit.revision, customConfigString });
+                    log.error(logContext, `Invalid config. Errors are ${parseResult.validationErrors.join(',')}`, { repoCloneUrl: commit.repository.cloneUrl, revision: commit.revision, customConfigString });
                 }
-                customConfig._origin = fromDefinitelyGp ? 'definitely-gp' : 'repo';
-            } else {
-                /* There is no configuration for this repository. Before we fall back to the default config,
-                * let's see if there is language specific configuration we might want to apply.
-                */
-                customConfig = await inferredConfig;
-                if (customConfig) {
-                    customConfig._origin = 'derived';
+            }
+            if (!customConfig) {
+
+                // try and find config file in the context repo or remote in
```

Comment:
for the reviewers: section below hasn't changed besides the indent because of the if.

Reply:
Tip: Adding `?w=1` at the end of the review URL ignores all whitespace changes (i.e. re-indented blocks will appear unchanged).